src/main/webapp/app/overview/course-statistics/course-statistics.component.ts

```diff
@@ -343,28 +344,82 @@ export class CourseStatisticsComponent implements OnInit, OnDestroy {
         return chartElement;
     }
 
-    // TODO: document the implementation of this method --> it is not really obvious
-    // TODO: save the return value of this method in the result object (as temp variable) to avoid that this method is invoked all the time
-    absoluteResult(result: Result): number | null {
+    generateTooltip(result: Result, groupedExercise: any): void {
         if (!result.resultString) {
-            return 0;
-        }
-        if (result.resultString && result.resultString.indexOf('failed') !== -1) {
-            return null;
+            groupedExercise.scores.tooltips.push(
+                this.translateService.instant('artemisApp.courseOverview.statistics.exerciseAchievedScore', {
+                    points: 0,
+                    percentage: 0,
+                }),
+            );
+            groupedExercise.missedScores.tooltips.push(
+                this.translateService.instant('artemisApp.courseOverview.statistics.exerciseMissedScore', {
+                    points: '',
+                    percentage: 100,
+                }),
+            );
         }
-        if (result.resultString && result.resultString.indexOf('passed') !== -1) {
-            return null;
+
+        const replaced = result.resultString.replace(',', '.');
+        const split = replaced.split(' ');
+
+        if (!replaced.includes('passed') && !replaced.includes('points')) {
+            if (result.score >= 50) {
+                groupedExercise.scores.tooltips.push(`${result.resultString} (${result.score}%)`);
+                groupedExercise.missedScores.tooltips.push(`(${100 - result.score}%)`);
+            } else {
+                groupedExercise.scores.tooltips.push(`(${result.score}%)`);
+                groupedExercise.missedScores.tooltips.push(`${result.resultString} (${100 - result.score}%)`);
+            }
+
+            return;
         }
-        if (result.submission && result.submission.submissionExerciseType === SubmissionExerciseType.PROGRAMMING && (result.submission as ProgrammingSubmission).buildFailed) {
-            return null;
+
+        if (replaced.includes('points')) {
+            if (split.length === 2) {
```

Comment:
Could you please document what split.length == 2/4 corresponds to (which exercisetype)?

Reply:
Done.